ext/adaptors/ranges-lib.tex

```diff
@@ -342,8 +352,8 @@
 \indexlibrary{\idxcode{at}!\idxcode{view_interface}}%
 \begin{itemdecl}
 template <ForwardRange C>
-  requires !View<C> && MoveConstructible<C> &&
-    ConvertibleTo<value_type_t<iterator_t<const D>>, value_type_t<iterator_t<C>>> &&
+  requires !View<C> && @\removed{MoveConstructible<C> \&\&}@
+    ConvertibleTo<@\changed{value_type_t}{reference_t}@<iterator_t<const D>>, value_type_t<iterator_t<C>>> &&
```

Comment:
Because of guaranteed copy elision?

Reply:
Yes. (it's in the commit log.)

Reply:
Clarifying: the `MoveConstructible` change is because of guaranteed copy elision, the `value_type_t` to `reference_t` change is to be a bit closer to the `EmplaceConstructible` requirement imposed by the Standard for range constructors of sequence containers.

Reply:
(Doing it properly would require checking allocator construction which is even less likely to be properly constrained.)